altair/vegalite/v5/theme.py

```diff
@@ -2,21 +2,33 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Final, Literal, get_args
+import sys
+from functools import wraps
+from typing import TYPE_CHECKING, Any, Callable, Dict, Final, Literal, TypeVar, get_args
 
 from altair.utils.theme import ThemeRegistry
 from altair.vegalite.v5.schema._typing import VegaThemes
 
-if TYPE_CHECKING:
-    import sys
+if sys.version_info >= (3, 10):
+    from typing import ParamSpec
+else:
+    from typing_extensions import ParamSpec
+
 
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 11):
+        from typing import LiteralString
+    else:
+        from typing_extensions import LiteralString
     if sys.version_info >= (3, 10):
         from typing import TypeAlias
     else:
         from typing_extensions import TypeAlias
 
+P = ParamSpec("P")
+R = TypeVar("R", bound=Dict[str, Any])
```

Comment:
The bounded `TypeVar` here can later be replaced with the `alt.Config` based `TypedDict` mentioned in https://github.com/vega/altair/issues/3519#issue-2447214476

While this currently provides *some* degree of type safety, the `TypedDict` solution will give detailed feedback on issues with the returned `dict`.